addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4112,7 +4112,7 @@ const SnippetOptionWidget = Widget.extend({
                 this.options.wysiwyg.odooEditor.historyStep();
             }
 
-            if (previewMode) {
+            if (previewMode || requiresReload) {
```

Comment:
> I'll merge this one like that to avoid the dead lock but this should really be better: if you don't reclick on the navbar after the first reload... you header is somehow in an inconsistent state? 

Clearly this flow is weird, but it was like that previously (worse even since clicking on the navbar would show a random popup with no context), so my aim was just to restore it.

It could be improved in two ways:
- Since reloading the editor can take a parameter to start on a specific snippet (see reloadSelector in the wysiwywg_adapter) we could find a way to pass the navbar selector so that it automatically clicks on it. It will still be done in 2 times (template selector, then no_hamburger selector) but it won't require user interaction
- Find a way for this to be done in backend, when a view is enabled, some CSS variables get activated/deactivated as well. If it's already possible, I'm not aware of it and it's my bad, this fix could have been better.

> following the flow you described, should not the no_hamburger_opt option be entirely hidden at the end? (that's why we had to disable it with the hack)... that's not the case, needs a fix 😬

Weird, in my tests it worked, I will investigate further.